src/api/objects/ObjectAPI.js

```diff
@@ -528,7 +527,9 @@ export default class ObjectAPI {
                 && provider.observe !== undefined
                 && this.SYNCHRONIZED_OBJECT_TYPES.includes(object.type)) {
                 let unobserve = provider.observe(identifier, (updatedModel) => {
-                    if (updatedModel.persisted > mutableObject.modified) {
+                    // modified can sometimes be undefined, so make it 0 in this case
+                    const mutableObjectModification = mutableObject.modified ? mutableObject.modified : 0;
```

Comment:
Perfect spot for the `??` operator:

```suggestion
                    const mutableObjectModification = mutableObject.modified ?? 0;
```